File: src/robot_basics/robot_basics/battery_monitor.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import Float32


class BatteryMonitor(Node):

    def __init__(self):
        super().__init__('battery_monitor')

        self.subscription = self.create_subscription(
            Float32,
            'battery_level',
            self.battery_callback,
            10
        )

    def battery_callback(self, msg):

        if msg.data <= 20:
            self.get_logger().warn(
                f'LOW BATTERY: {msg.data:.1f}%'
            )
        else:
            self.get_logger().info(
                f'Battery OK: {msg.data:.1f}%'
            )


def main(args=None):
    rclpy.init(args=args)

    node = BatteryMonitor()

    rclpy.spin(node)

    node.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()

File: src/robot_basics_cpp/src/battery_monitor.cpp
#include <memory>

#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/float32.hpp"

class BatteryMonitor : public rclcpp::Node
{
public:
  BatteryMonitor()
  : Node("battery_monitor_cpp")
  {
    subscription_ = this->create_subscription<std_msgs::msg::Float32>(
      "/battery_level",
      10,
      std::bind(&BatteryMonitor::battery_callback, this, std::placeholders::_1)
    );
  }

private:
  void battery_callback(const std_msgs::msg::Float32::SharedPtr msg)
  {
    RCLCPP_INFO(
      this->get_logger(),
      "Battery level: %.2f",
      msg->data
    );
  }

  rclcpp::Subscription<std_msgs::msg::Float32>::SharedPtr subscription_;
};

int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<BatteryMonitor>());
  rclcpp::shutdown();

  return 0;
}
